### PR TITLE
fix: Add Big Patent Classification dataset

### PR DESCRIPTION
**Pull request by shreeya-dhakal**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/eng/PatentClassification.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        form=None,
+        domains=[],
+        task_subtypes=None,
+        license="Not specified",
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=[],
+        text_creation=None,
+        bibtex_citation=None,
+        n_samples={"test": 5000},
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/PatentClassification.py:32]: please subsample this to at most 2048, according to the PR checklist.

**Review by isaac-chung — CHANGES_REQUESTED** [review]
Thanks for adding this! I have the same comments as in https://github.com/embeddings-benchmark/mteb/pull/479.

**Comment by shreeya-dhakal** [comment]
> Thanks for adding this! I have the same comments as in #479.

@isaac-chung I have added the metadata.

File: mteb/tasks/Classification/eng/PatentClassification.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        date=("2021-11-05", "2022-10-22"),
+        form=["written"],
+        domains=["Legal"],
+        task_subtypes=["Topic classification"],
+        license="Not specified",
+        socioeconomic_status="high",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/PatentClassification.py:30]:
```suggestion
        bibtex_citation="""@inproceedings{sharma-etal-2019-bigpatent,
    title = "{BIGPATENT}: A Large-Scale Dataset for Abstractive and Coherent Summarization",
    author = "Sharma, Eva  and
      Li, Chen  and
      Wang, Lu",
    editor = "Korhonen, Anna  and
      Traum, David  and
      M{\`a}rquez, Llu{\'\i}s",
    booktitle = "Proceedings of the 57th Annual Meeting of the Association for Computational Linguistics",
    month = jul,
    year = "2019",
    address = "Florence, Italy",
    publisher = "Association for Computational Linguistics",
    url = "https://aclanthology.org/P19-1212",
    doi = "10.18653/v1/P19-1212",
    pages = "2204--2213",
    abstract = "Most existing text summarization datasets are compiled from the news domain, where summaries have a flattened discourse structure. In such datasets, summary-worthy content often appears in the beginning of input articles. Moreover, large segments from input articles are present verbatim in their respective summaries. These issues impede the learning and evaluation of systems that can understand an article{'}s global content structure as well as produce abstractive summaries with high compression ratio. In this work, we present a novel dataset, BIGPATENT, consisting of 1.3 million records of U.S. patent documents along with human written abstractive summaries. Compared to existing summarization datasets, BIGPATENT has the following properties: i) summaries contain a richer discourse structure with more recurring entities, ii) salient content is evenly distributed in the input, and iii) lesser and shorter extractive fragments are present in the summaries. Finally, we train and evaluate baselines and popular learning models on BIGPATENT to shed light on new challenges and motivate future directions for summarization research.",
}""",
```

**Review by isaac-chung — COMMENTED** [review]
Almost there! Just a small comment.
And please add 2 pts for the dataset and 2 pts for my review.

**Review by isaac-chung — APPROVED** [review]
Looks good! Please add 2 pts for the dataset and 2 pts for my review.

**Comment by shreeya-dhakal** [comment]
> Looks good! Please add 2 pts for the dataset and 2 pts for my review.

@isaac-chung done!

**Pull request merged**